x/datapool/keeper/pool.go

```diff
@@ -169,12 +177,38 @@ func (k Keeper) CreatePool(ctx sdk.Context, curator sdk.AccAddress, poolParams t
 		return 0, sdkerrors.Wrapf(err, "failed to marshal mint NFT msg")
 	}
 
-	_, err = k.wasmKeeper.Execute(ctx, contractAddr, types.GetModuleAddress(), mintMsgBz, zeroFund)
+	moduleAddr := types.GetModuleAddress()
+
+	_, err = k.wasmKeeper.Execute(ctx, nftContractAddr, moduleAddr, mintMsgBz, zeroFund)
 
 	if err != nil {
 		return 0, sdkerrors.Wrapf(err, "failed to mint curator NFT")
 	}
 
+	poolName := "pool_" + strconv.FormatUint(newPool.GetPoolId(), 10)
+	symbol := "DATA" + strconv.FormatUint(newPool.GetPoolId(), 10)
+
+	instantiateMsg := types.NewInstantiateNFTMsg(poolName, symbol, newPoolAddr)
+	instantiateMsgBz, err := json.Marshal(instantiateMsg)
+	if err != nil {
+		return 0, sdkerrors.Wrapf(err, "failed to marshal instantiateMsg")
+	}
+
+	codeID := k.GetParams(ctx).DataPoolCodeId
+
+	poolAccAddr, err := sdk.AccAddressFromBech32(newPoolAddr)
```

Comment:
I think it chaged `sdk` to `types`
```suggestion
	poolAccAddr, err := types.AccAddressFromBech32(newPoolAddr)
```

Reply:
Ah, I checked and it doesn't seem to be used when creating the module address.
You don't need to change this.

Reply:
There is a part that gets the same poolAddress on line 130 (duplicate code)

Reply:
Thank you. It is duplicated and confused code. I will fix it

Reply:
fixed in [49b64b5](https://github.com/medibloc/panacea-core/pull/303/commits/49b64b5401217524872f5f612a5bd52775b51d0a)

Reply:
Good